File: scratch/count_chars.py
import yaml
import json
with open('../config/tools.yaml', encoding='utf-8') as f:
    data = yaml.safe_load(f)
for t in data['tools']:
    if t['slug'] == 'redirect-checker':
        def print_lengths(obj, path=""):
            if isinstance(obj, dict):
                for k, v in obj.items():
                    print_lengths(v, path + "." + k)
            elif isinstance(obj, list):
                for i, v in enumerate(obj):
                    print_lengths(v, path + "[" + str(i) + "]")
            elif isinstance(obj, str):
                print(f"{path}: {len(obj)}")
        print_lengths(t)

File: scratch/count_chars.js
const fs = require('fs');
const yaml = require('js-yaml');
const file = fs.readFileSync('../config/tools.yaml', 'utf8');
const data = yaml.load(file);
const t = data.tools.find(x => x.slug === 'redirect-checker');

function printLengths(obj, path = "") {
    if (typeof obj === 'object' && obj !== null) {
        if (Array.isArray(obj)) {
            obj.forEach((v, i) => printLengths(v, path + "[" + i + "]"));
        } else {
            for (const [k, v] of Object.entries(obj)) {
                printLengths(v, path + "." + k);
            }
        }
    } else if (typeof obj === 'string') {
        console.log(path + ": " + obj.length);
    }
}
printLengths(t);
